source/extensions/filters/http/ext_proc/matching_utils.cc

```diff
@@ -0,0 +1,102 @@
+#include "source/extensions/filters/http/ext_proc/matching_utils.h"
+
+#include <memory>
+
+#if defined(USE_CEL_PARSER)
+#include "parser/parser.h"
+#endif
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace ExternalProcessing {
+
+absl::flat_hash_map<std::string, ExpressionManager::CelExpression>
+ExpressionManager::initExpressions(const Protobuf::RepeatedPtrField<std::string>& matchers) {
+  absl::flat_hash_map<std::string, ExpressionManager::CelExpression> expressions;
+#if defined(USE_CEL_PARSER)
+  for (const auto& matcher : matchers) {
+    if (expressions.contains(matcher)) {
+      continue;
+    }
+    auto parse_status = google::api::expr::parser::Parse(matcher);
+    if (!parse_status.ok()) {
+      throw EnvoyException("Unable to parse descriptor expression: " +
+                           parse_status.status().ToString());
+    }
+
+    Filters::Common::Expr::ExpressionPtr expression =
+        Extensions::Filters::Common::Expr::createExpression(builder_->builder(),
+                                                            parse_status.value().expr());
+
+    expressions.emplace(
+        matcher, ExpressionManager::CelExpression{parse_status.value(), std::move(expression)});
+  }
+#else
+  ENVOY_LOG(warn, "CEL expression parsing is not available for use in this environment."
+                  " Attempted to parse " +
+                      std::to_string(matchers.size()) + " expressions");
+#endif
+  return expressions;
+}
+
+ProtobufWkt::Struct
+ExpressionManager::evaluateAttributes(const Filters::Common::Expr::Activation& activation,
+                                      const absl::flat_hash_map<std::string, CelExpression>& expr) {
+
+  ProtobufWkt::Struct proto;
+
+  if (expr.empty()) {
+    return proto;
+  }
+
+  for (const auto& hash_entry : expr) {
+    ProtobufWkt::Arena arena;
+    const auto result = hash_entry.second.compiled_expr_->Evaluate(activation, &arena);
+    if (!result.ok()) {
+      // TODO: Stats?
+      continue;
+    }
+
+    if (result.value().IsError()) {
+      ENVOY_LOG(trace, "error parsing cel expression {}", hash_entry.first);
+      continue;
+    }
+
+    ProtobufWkt::Value value;
+    switch (result.value().type()) {
+    case google::api::expr::runtime::CelValue::Type::kBool:
+      value.set_bool_value(result.value().BoolOrDie());
+      break;
+    case google::api::expr::runtime::CelValue::Type::kNullType:
+      value.set_null_value(ProtobufWkt::NullValue{});
+      break;
+    case google::api::expr::runtime::CelValue::Type::kDouble:
+      value.set_number_value(result.value().DoubleOrDie());
+      break;
+    default:
+      value.set_string_value(Filters::Common::Expr::print(result.value()));
+    }
+
+    auto proto_mut_fields = proto.mutable_fields();
+    (*proto_mut_fields)[hash_entry.first] = value;
+  }
+
+  return proto;
+}
+
+const std::vector<Matchers::StringMatcherPtr>
```

Comment:
nit : const in return type here is effectively redundant and it potentially prevent the compiler from optimizing
I think effectively `const` is used when returning pointers or references to const objects.

Reply:
Good call; removed this `const`